transformations/replace_abbreviation_and_acronyms/transformation.py

```diff
@@ -0,0 +1,309 @@
+    ```
+    tokens = [{"text": "as", ...}, {"text": "soon", ...}, {"text": "as", ...}, {"text": "possible", ...}, {"text": "do", ...}]
+    abbreviation = "as soon as possible"
+    (0, 3) = identify_init_and_end_of_expanded_abbreviations(tokens, abbreviation)
+    ```
+
+    Parameters:
+        `tokens`: The list of tokens representing a sentence.
+        `abbreviation`: The abbreviation string to search for in the list of tokens.
+    Returns:
+        The first and last index of abbreviation if found in the token list.
+    """
+    candidate_abbreviation: str = ""
+    for i, token in enumerate(tokens):
+        candidate_abbreviation = (
+            candidate_abbreviation + token["text"] + token["end_space"]
+        )
+        if candidate_abbreviation.lower().strip() == abbreviation.lower():
+            return 0, i
+        if candidate_abbreviation.lower() in abbreviation.lower():
+            continue
+        else:
+            return -1, -1
+    return -1, -1
+
+
+def fuse_token_values_for_expanded_abbreviations(
+    tokens: List, merge_collection: Set[Tuple[int, int]]
+) -> List:
+    """
+    Fuse token values for tokens that have been identified as expanded abbreviations into a single token.
+    Ex: `[{'text' : 'as'}, {'text' : 'soon'}, {'text' : 'as'}, {'text' : 'possible'}]` =>
+    `[{'text' : 'as soon as possible'}]`.
+
+    Parameters:
+        `tokens`: The list of tokens representing a sentence.
+        `merge_collection`: List of `start` and `finish` indexes.
+            `start`: The first index of identified abbreviation.
+            `finish`: The last index of identified abbreviation.
+    Returns:
+        A list of tokens with expanded abbreviation tokens fused.
+    """
+    start_indexes = [start for start, _ in merge_collection]
+    new_tokens = []
+    counter = 0
+    while counter < len(tokens):
+        if counter not in start_indexes:
+            new_tokens.append(tokens[counter])
+            counter += 1
+            continue
+
+        finish = [
+            finish for start, finish in merge_collection if start == counter
+        ][0]
+
+        if finish - counter == 0:
+            tokens[counter]["is_abbreviation"] = True
+            tokens[counter]["is_expanded_abbreviation"] = True
+            new_tokens.append(tokens[counter])
+            counter += 1
+            continue
+
+        new_token_text = {}
+
+        text = "".join(
+            [
+                token["text"] + token["end_space"]
+                for token in tokens[counter : finish + 1]
+            ]
+        )
+        new_token_text["text"] = text.strip()
+        new_token_text["end_space"] = tokens[finish]["end_space"]
+        new_token_text["is_abbreviation"] = True
+        new_token_text["is_expanded_abbreviation"] = True
+
+        new_tokens.append(new_token_text)
+        counter = finish + 1
+
+    return new_tokens
+
+
+class ReplaceAbbreviations(SentenceOperation):
```

Comment:
Could you please add docstrings for class also?

Reply:
done